File: Task-001/Task1.scala
import scala.io.Source

object Task1 {
	// takes in a string text and outputs a mutable map[string int]
	def wordCount(text : String): scala.collection.mutable.Map[String, Int] = {
		// create map
		var wordCountMap = scala.collection.mutable.Map[String, Int]()
		// test data
		wordCountMap("hello") = 1
		// turn all into lowercase
		val text_lowercase = text.toLowerCase()
		// turn string into words not characters
		val words = text_lowercase.split(" ")


		for (i <- words){
			if (wordCountMap.contains(i)){
				wordCountMap(i) += 1

			}
			else{
				wordCountMap(i) = 1 
			}

		}
		return wordCountMap
	}

	def main(args: Array[String]){

	val source = Source.fromFile("Shakespeare.txt")
	val lines = source.getLines
	var text = " "
	var num = 1
	while(lines.hasNext){
		text = text + lines.next
		num += 1
		println(num + " out of 124457")

	}
	source.close
	text = text.replaceAll("[-+.^:,*]"," ");
	println(wordCount(text))

	






		
	}
}

File: Task-001/task1.py
import re
def word_count():
	
	# load text
	text = open("Shakespeare.txt", "r")
	
	# turn text into a string
	text_to_string = " "
	for line in text:
		text_to_string= text_to_string + " " + line

	# get rid of all symbols using regex 
	text_only = re.sub(r'[^\w]', ' ', text_to_string)


	# split text into words
	split_text = text_only.split(" ")


	word_dictionary= {}

	for word in split_text:
		# check if the word is already added into dictionary

		if word.lower() in word_dictionary:
			# increment value 
			word_dictionary[word.lower()] = word_dictionary[word.lower()] +1
		# dont add if not a word
		elif word == "" :
			continue
		# add into dictionary if it isnt there and set value to 1 
		else:
			word_dictionary[word.lower()] = 1

	text.close()
	for i in word_dictionary:
		print (i, " ", word_dictionary[i])



word_count()





